### PR TITLE
docker: remove unused build tools

### PR DESCRIPTION
**Pull request by pujan14**
Signed-off-by: Pujan Shah <pujan14@gmail.com>

Removed unused build tools from dockerfile and added default argument to image with CMD

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,7 +12,14 @@ LABEL maintainer=terraform-linters
 
 RUN apk add --no-cache ca-certificates
 
-COPY --from=builder /tflint/dist/tflint /usr/local/bin
+# copy tflint from builder
```
Comment (bendrucker) [inline Dockerfile:15]: Remove these comments, they are unnecessarily re-stating things that can be readily inferred from the code. Everything happening here is basic and well-documented upstream. You can assume basic understanding of Dockerfiles here and not try to explain how they work.

File: Dockerfile [hunk 2]
```diff
@@ -12,7 +12,14 @@ LABEL maintainer=terraform-linters
...
+COPY --from=builder /tflint/dist/tflint /usr/local/bin/tflint
 
-ENTRYPOINT ["tflint"]
+# set working directory
 WORKDIR /data
+
+# set the default entrypoint -- when this container is run, use this command
+ENTRYPOINT [ "tflint" ]
```
Comment (bendrucker) [inline Dockerfile:22]: Keep the formatting consistent with Docker's docs, meaning no array padding:

https://docs.docker.com/engine/reference/builder/

File: Dockerfile [hunk 3]
```diff
@@ -12,7 +12,14 @@ LABEL maintainer=terraform-linters
...
+
+# as we specified an entrypoint, this is appended as an argument (i.e., `tflint --help`)
+CMD [ "--help" ]
```
Comment (bendrucker) [inline Dockerfile:25]: Is this really the best default command, as opposed to invoking TFLint with no arguments, which is implicitly the same as `tflint .`?

**Review by bendrucker — CHANGES_REQUESTED** [review]
I don't agree with the user-facing change proposed here, but if those APK packages in the builder phase are unnecessary, happy to accept that change on its own.

File: Dockerfile (hunk 3 above)
Comment (pujan14) [reply Dockerfile:25]: reverting back to original behaviour

**Pull request merged**